### PR TITLE
Header saving

### PR DESCRIPTION
**Pull request by sawer297**
added code that will write the request and response headers to a file if save-all is set to true
also updated the requests to use a custom user-agent

### CODE REVIEW

File: opendap_tests.py [hunk 1]
```diff
@@ -38,7 +38,10 @@ def dmr_tester(url_address):
     results = {"dmr_test": tr}
     try:
         print(".", end="", flush=True) if not quiet else False
-        r = requests.get(url_address + ext)
+
+        headers = requests.utils.default_headers()
```
Comment (jgallagher59701) [inline opendap_tests.py]: This two line bit could be put in a function and then the call to requests.get() could be "r = requests.get(url_address + ext, headers=pydmr_headers())" That way we would not have to update the string in three places. Not a huge deal, but worth thinking about.

File: opendap_tests.py [hunk 2]
```diff
@@ -171,6 +179,14 @@ def save_response(url_address, ext, r):
         base_name = save + '/' + base_name
     with open(base_name + ext, 'w') as file:
         file.write(r.text)
+    with open(base_name + ext + ".h", 'w') as header:
```
Comment (jgallagher59701) [inline opendap_tests.py:180]: I think we should keep this in the code. If we are saving the response, lets also keep the headers. Good call.

**Review by jgallagher59701 — APPROVED** [review]
This can be merged, but I think factoring those two lines into a single function would be an improvement.

File: opendap_tests.py (hunk 1 above)
Comment (sawer297) [reply opendap_tests.py]: good point, I can knock this out real quick

Comment (sawer297) [reply opendap_tests.py]: finally fixed after my laptop froze and crashed

**Pull request merged**